### PR TITLE
V22F-925 Fix racecondition

### PR DESCRIPTION
**Pull request by KungFuDonkey**
(Hopefully) fixes the race condition by setting socket options and moving the state of the receiving thread to an integer

### CODE REVIEW

File: IPCLib/Socket.cpp [hunk 1]
```diff
@@ -20,40 +43,48 @@ ReceivingThread::ReceivingThread(const std::function<void(bool*)>& p_receiveData
...
 }
 
-/// @brief  Checks whether the thread started receiving
-/// @return Whether the thread started receiving
+/// @brief  Check whether the thread has started receiving
+/// @return Whether the thread has started receiving
 bool ReceivingThread::StartedReceiving() const
 {
-    return m_startedReceiving;
+    return GET_STARTED_RECEIVING() || GET_RECEIVED();
```
Comment (noutheesink) [inline IPCLib/Socket.cpp:61]: Started receiving will be true if GET_RECEIVED() is true? Is this what you need? `GET_STARTED_RECEIVING() || GET_RECEIVING()` makes more sense to me

**Review by noutheesink — CHANGES_REQUESTED** [review]

File: IPCLib/Socket.cpp (hunk 1 above)
Comment (KungFuDonkey) [reply IPCLib/Socket.cpp:61]: no the condition should be true if it started or has received a message, as both state that the worker thread started running but we do not know at what tume we are checking

**Review by noutheesink — APPROVED** [review]

File: IPCLib/Socket.cpp [hunk 2]
```diff
@@ -3,15 +3,38 @@
...
+#define STARTED_RECEIVING_STATE 0b00001000
+#define RECEIVED_STATE          0b00000100
+#define RECEIVING_STATE         0b00000010
+#define STOP_STATE              0b00000001
+
+#define GET_ERROR()             ((m_state & ERROR_STATE) > 0)
+#define GET_STARTED_RECEIVING() ((m_state & STARTED_RECEIVING_STATE) > 0)
+#define GET_RECEIVED()          ((m_state & RECEIVED_STATE) > 0)
+#define GET_RECEIVING()         ((m_state & RECEIVING_STATE) > 0)
+#define GET_STOP()              ((m_state & STOP_STATE) > 0)
```
Comment (suppergerrie2) [inline IPCLib/Socket.cpp:23]: Should this be != instead of = so it works on signed ints?

**Review by suppergerrie2 — CHANGES_REQUESTED** [review]

File: IPCLib/Socket.cpp (hunk 2 above)
Comment (KungFuDonkey) [reply IPCLib/Socket.cpp:23]: what? m_state is an unsinged int? there is no = here?

**Review by suppergerrie2 — APPROVED** [review]

**Pull request merged**